packages/synapse-sdk/src/telemetry/service.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * TelemetryService - Main telemetry service for Synapse SDK
+ *
+ * You should use `synapse.telemetry.sentry` directly to capture events.
+ *
+ * ## Debug Dumps
+ *
+ * Get recent events for support tickets:
+ *
+ * ```typescript
+ * const dump = synapse.telemetry.debugDump()
+ * console.log(JSON.stringify(dump, null, 2))
+ * ```
+ */
+
+import type { BrowserOptions, ErrorEvent, EventHint } from '@sentry/browser'
+import type { NodeOptions } from '@sentry/node'
+import type { FilecoinNetworkType } from '../types.ts'
+import { SDK_VERSION } from '../utils/sdk-version.ts'
+import { getSentry, isBrowser, type Sentry as SentryType } from './utils.ts'
+
+type SentryInitOptions = BrowserOptions | NodeOptions
+type SentrySetTags = Parameters<SentryType['setTags']>[0]
+
+export interface TelemetryConfig {
+  /**
+   * Additional options to pass to the Sentry SDK's init method.
+   */
+  sentryInitOptions?: SentryInitOptions
+  /**
+   * Additional tags to set on the Sentry SDK.
+   */
+  sentrySetTags?: SentrySetTags
+}
+
+/**
+ * Configuration for runtime detection and context
+ */
+export interface TelemetryRuntimeContext {
+  filecoinNetwork: FilecoinNetworkType
+}
+
+export interface DebugDump {
+  events: any[]
+}
+
+/**
+ * Main telemetry service that manages the adapter and provides high-level APIs
+ */
+export class TelemetryService {
+  private config: TelemetryConfig
+  private context: TelemetryRuntimeContext
+  private eventBuffer: any[] = []
+  private readonly maxBufferSize = 50
+
+  sentry: SentryType | null = null
+
+  constructor(config: TelemetryConfig, context: TelemetryRuntimeContext) {
+    this.context = context
+    this.config = config
+
+    // Initialize sentry always.. singleton.ts will not construct this service if telemetry is disabled.
+    void this.initSentry()
+  }
+
+  private async initSentry(): Promise<void> {
```

Comment:
```suggestion
  // This is a separate function rather than being in the constructor because it is async.
  // This does means that a TelemetryService instance can be accessible without the sentry object being instantiated.
  // We are fine with this in practice because in the worst case it means some initial telemetry events get missed.
  // Consuming code of the Synapse telemetry module should be fine because it already protects against a null sentry instance in case telemetry is disabled. 
  private async initSentry(): Promise<void> {
```

Ignore if this is generally obvious.